t/rest_server/.github/workflows/deploy.yaml

```diff
@@ -0,0 +1,127 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: 'deploy'
+
+on:
+  workflow_dispatch:
+    inputs:
+      deploymentEnvironment:
+        type: 'choice'
+        description: 'Environment to deploy image to.'
+        default: 'autopush'
+        options:
+          - 'autopush'
+  workflow_call:
+    inputs:
+      deploymentEnvironment:
+        type: 'string'
+        description: 'Environment to deploy image to.'
+        default: 'autopush'
+
+concurrency:
+  group: '${{ github.workflow }}-${{ github.head_ref || github.ref }}'
+  cancel-in-progress: true
+
+env:
+  # This should be the only env variable defined in this file. Define all other
+  # variables in the bootstrap file.
+  # More environments are expected to be added in the future.
+  AUTOPUSH_ENV_FILE: '.github/config/autopush.env'
+  PRODUCTION_ENV_FILE: '.github/config/production.env'
+
+jobs:
+  image-release:
+    permissions:
+      contents: 'read'
+      id-token: 'write'
+    name: 'Release Image'
+    runs-on: 'ubuntu-latest'
+    steps:
+      - name: 'Checkout Repo'
+        uses: 'actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab' # ratchet:actions/checkout@v3
+      - name: 'Load variables from Autopush .env file'
+        if: "${{ inputs.deploymentEnvironment == 'autopush' }}"
+        uses: 'abcxyz/pkg/.github/actions/override-workflow-variables@main'
+        with:
+          env_file: '${{ env.AUTOPUSH_ENV_FILE }}'
+      - id: 'auth'
+        name: 'Authenticate to Google Cloud'
+        uses: 'google-github-actions/auth@e8df18b60c5dd38ba618c121b779307266153fbf' # ratchet:google-github-actions/auth@v1
+        with:
+          token_format: 'access_token'
+          service_account: '${{ env.AUTOMATION_SERVICE_ACCOUNT }}'
+          workload_identity_provider: '${{ env.WIF_PROVIDER }}'
+      - name: 'Login to Artifact Registry'
+        uses: 'docker/login-action@f4ef78c080cd8ba55a85445d5b36e214a81df20a' # ratchet:docker/login-action@v2
+        with:
+          registry: '${{ env.AR_LOCATION }}'
+          username: 'oauth2accesstoken'
+          password: '${{ steps.auth.outputs.access_token }}'
+      - name: 'Setup Go'
+        uses: 'actions/setup-go@4d34df0c2316fe8122ab82dc22947d607c0c91f9' # ratchet:actions/setup-go@v4
+        with:
+          go-version: '1.20.2'
+      - name: 'Build Go binary'
+        run: |
+          CGO_ENABLED=0 GOOS=linux go build -o restserver .
```

Comment:
Is the GOOS necessary? SInce this is running on ubuntu-latest, it will automatically build for Linux. We don't seem to do this elsewhere: https://github.com/search?q=org%3Aabcxyz%20%22go%20build%22&type=code

Reply:
Removed GOOS, unsure if it's needed.